src/blas/backends/portblas/portblas_level1.cxx

```diff
@@ -299,6 +305,11 @@ sycl::event nrm2(sycl::queue &queue, std::int64_t n, const std::complex<real_t>
 
 sycl::event nrm2(sycl::queue &queue, std::int64_t n, const real_t *x, std::int64_t incx,
                  real_t *result, const std::vector<sycl::event> &dependencies) {
+    // portBLAS nrm2 implementation requires result to be initializes to zero
+    // before starting the computation.
+    auto init_res_val = queue.submit(
+        [&](sycl::handler &cgh) { cgh.single_task([=]() { result[0] = real_t(0); }); });
+    init_res_val.wait();
```

Comment:
Instead of waiting on this event here, I think we can add this event to the `dependencies` vector and that should take care of things.